### ISSUE TITLE
out_prometheus_remote_write: missing timestamp in metrics

### ISSUE THREAD

**Issue by edsiper** [fluent/fluent-bit#5304]
## Bug Report

Prometheus remote write output plugin is not setting the timestamp correctly. 

Originally reported on Slack community channel:

- https://fluent-all.slack.com/archives/C0CTQGHKJ/p1649692991347309

## Repro case

1. Start Prometheus Server
2. Start Prometheus node exporter (listens on 9100)
3. Start Fluent Bit to scrape metrics from Prometheus node exporter:

```
fluent-bit -i prometheus_scrape -p scrape_interval=2 -p 9100 \
           -o prometheus_remote_write -p port=9090 -p uri="/api/v1/write"  -f 1 -v
```

Prometheus Server will start printing the errors in stdout:

```
ts=2022-04-13T03:06:15.320Z caller=write_handler.go:102 level=error component=web msg="Out of order sample from remote write" err="out of bounds" series="{__name__=\"go_gc_duration_seconds_count\"}" timestamp=0
ts=2022-04-13T03:06:17.325Z caller=write_handler.go:102 level=error component=web msg="Out of order sample from remote write" err="out of bounds" series="{__name__=\"go_gc_duration_seconds_count\"}" timestamp=0
ts=2022-04-13T03:06:19.311Z caller=write_handler.go:102 level=error component=web msg="Out of order sample from remote write" err="out of bounds" series="{__name__=\"go_gc_duration_seconds_count\"}" timestamp=0
ts=2022-04-13T03:06:19.312Z caller=write_handler.go:102 level=error component=web msg="Out of order sample from remote write" err="out of bounds" series="{__name__=\"go_gc_duration_seconds_count\"}" timestamp=0
ts=2022-04-13T03:06:20.329Z caller=write_handler.go:102 level=error component=web msg="Out of order sample from remote write" err="out of bounds" series="{__name__=\"go_gc_duration_seconds_count\"}" timestamp=0
ts=2022-04-13T03:06:20.329Z caller=write_handler.go:102 level=error component=web msg="Out of order sample from remote write" err="out of bounds" series="{__name__=\"go_gc_duration_seconds_count\"}" timestamp=0
ts=2022-04-13T03:06:21.323Z caller=write_handler.go:102 level=error component=web msg="Out of order sample from remote write" err="out of bounds" series="{__name__=\"go_gc_duration_seconds_count\"}" timestamp=0
```

note that __timestamp=0__

Fluent Bit logs:

```
[2022/04/12 21:08:23] [error] [output:prometheus_remote_write:prometheus_remote_write.0] 127.0.0.1:9090, HTTP status=400
out of bounds
```

**Comment by leonardo-albertovich** [comment]
The problem in this case seems to be that the node exporter exposes a summary which is empty until the first garbage collection pass happens. Because there is no clear indication of this we ingest it the way it is setting timestamp to zero but this is not what their own parser does, [their parser](https://github.com/prometheus/prometheus/blob/main/scrape/scrape.go#L1489) replaces those with the current timestamp (unless I misunderstood it).

### Sample data (obtained with `curl http://127.0.0.1:9100/metrics | grep go_gc`)

```
go_gc_duration_seconds{quantile="0"} 0
go_gc_duration_seconds{quantile="0.25"} 0
go_gc_duration_seconds{quantile="0.5"} 0
go_gc_duration_seconds{quantile="0.75"} 0
go_gc_duration_seconds{quantile="1"} 0
go_gc_duration_seconds_sum 0
go_gc_duration_seconds_count 0
```

### Sample data after being processed by Opentelemetrys prometheus scraper (which uses the original prometheus code)

```
              "name": "go_gc_duration_seconds",
              "summary": {
                "dataPoints": [
                  {
                    "startTimeUnixNano": "1649961270899000000",
                    "timeUnixNano": "1649961270899000000",
                    "quantileValues": [
                      {},
                      {
                        "quantile": 0.25
                      },
                      {
                        "quantile": 0.5
                      },
                      {
                        "quantile": 0.75
                      },
                      {
                        "quantile": 1
                      }
                    ]
                  }
                ]
              }
            },
```

This, and a quick `date +%s%N` suggest that the previous statement is true.
In that case we might want to implement that logic in the prometheus decoder if we want to mimic their behavior, otherwise, we could implement it in the encoder but I would be careful of the implications.

**Comment by leonardo-albertovich** [comment]
I was wrong after all, the real issue was in the prometheus decoder, my assumption that there would be a timestamp added to the data after garbage collection happened was wrong, the node exporter does not add those timestamps.

**Comment by edsiper** [comment]
Fixed with #5321
